templatestore/frontend/src/components/TinyUrlComponent.js

```diff
@@ -0,0 +1,172 @@
+            if (id !== "") {
+                visitedCopy[id] = 0;
+            }
+            props.updateVisited(visitedCopy);
+        }
+        let index = newItems.findIndex(item => item.urlKey === id);
+        newItems[index][e.target.name] = e.target.value;
+        props.updateItems(newItems);
+    }
+
+    function arrayChanges() {
+        const isSame = (a, b) => a.urlKey === b.urlKey && a.expiry === b.expiry;
+        // Get props.items that only occur in the left array,
+        // using the compareFunction to determine equality.
+        const onlyInLeft = (left, right, compareFunction) =>
+            left.filter(leftValue =>
+                !right.some(rightValue =>
+                    compareFunction(leftValue, rightValue)));
+        let addedArray, removedArray;
+        if (tinyUrlObjLocal) {
+            removedArray = onlyInLeft(tinyUrlObjLocal, props.items, isSame);
+            addedArray = onlyInLeft(props.items, tinyUrlObjLocal, isSame);
+        } else {
+            removedArray = [];
+            addedArray = props.items;
+        }
+        return { itemsRemoved: removedArray,
+            itemsAdded: addedArray };
+    }
+
+    function valueSelected(e) {
+        e.preventDefault();
+        let data = {};
+        for (let i = 0; i < props.items.length; i++) {
+            if (props.items[i].urlKey === "" || props.items[i].expiry === "") {
+                props.showAlerts("Can't leave url or expiry blank");
+                return;
+            }
+        }
+        let changes = arrayChanges();
+        if (changes.itemsAdded.length === 0 && changes.itemsRemoved.length === 0) {
+            props.showAlerts("Please modify something");
+            return;
+        }
+
+        for (let i = 0; i < changes.itemsRemoved.length; i++) {
+            const index = tinyUrlObjLocal.findIndex(obj => obj.urlKey === changes.itemsRemoved[i].urlKey); // eslint-disable-line
+            if (index !== -1)tinyUrlObjLocal.splice(index, 1);
+        }
+        if (changes.itemsAdded.length) {
+            if (tinyUrlObjLocal)tinyUrlObjLocal = [...tinyUrlObjLocal, ...changes.itemsAdded];
+            else tinyUrlObjLocal = changes.itemsAdded;
+        }
+        data.tinyUrlArray = tinyUrlObjLocal;
+        data.templateName = props.templateName ? props.templateName : "";
+        data.templateVersion = props.templateVersion ? props.templateVersion : "";
+        axios({
+            method: 'put',
+            url: backendSettings.TE_BASEPATH + '/api/v1/tiny_url',
+            data: data
+        }).then((response) => {
+            props.showAlerts(response.data.message);
+        }).catch((err) => {props.showAlerts(err.response.data);});
+    }
+
+    return (
+        <>
+            <div className={styles.teMarginTop20}>
+                <label>TinyUrl: </label>
+            </div>
+            <div
+                className={styles.teAccordian + ' accordion md-accordion'}
+                id="accordionEx2"
```

Comment:
Are we uniquely identifying any fields via `id`? If not, let's remove them all.

Reply:
Used previous styling which is already defined.